src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -105,15 +112,9 @@ public KafkaTopicRepository(final ZooKeeperHolder zkFactory, final KafkaFactory
     public void createTopic(final EventType eventType) throws TopicCreationException, DuplicatedEventTypeNameException {
         createTopic(eventType.getTopic(),
                 calculateKafkaPartitionCount(eventType.getDefaultStatistic()),
-                settings.getDefaultTopicReplicaFactor(),
-                getTopicRetentionMs(eventType),
-                settings.getDefaultTopicRotationMs());
-    }
-
-    private long getTopicRetentionMs(final EventType eventType) {
-        return Optional.ofNullable(eventType.getOptions())
-                .map(EventTypeOptions::getRetentionTime)
-                .orElse(settings.getDefaultTopicRetentionMs());
+                nakadiSettings.getDefaultTopicReplicaFactor(),
+                eventType.getOptions().getRetentionTime(),
```

Comment:
But it can not be null here we set to default value in the controller (the whole feature is about it). If somebody is going to call this method separately then it is the responsibility of the caller. Also I think we will have some kind of exception when we provide 'null' to kafka config (I have not checked it though). Also we talked to @rcillo about it.